### PR TITLE
Convert users and variables types to knex

### PR DESCRIPTION
**Pull request by danyx23**
Adds knex based DB access code for tables users and variables

### CODE REVIEW

**Comment by danyx23** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3145?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #3054** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3054?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3056** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3056?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3064** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3064?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3145** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3145?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
          * **PR #3074** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3074?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3065** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3065?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3143** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3143?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #3131** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3131?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #3139** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3139?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3145?utm_source=stack-comment).

File: db/model/Dataset.ts [hunk 1]
```diff
@@ -90,10 +106,11 @@ export class Dataset extends BaseEntity {
...
         const sources = await Source.findBy({ datasetId: this.id })
         const variables = (await db
-            .knexTable(variableTable)
-            .where({ datasetId: this.id })) as VariableRow[]
-        const tags = await db.queryMysql(
+            .knexTable(VariablesTableName)
+            .where({ datasetId: this.id })) as DbRawVariable[]
+        const tags: Pick<DbPlainTag, "id" | "name">[] = await knexRaw(
             `SELECT t.id, t.name FROM dataset_tags dt JOIN tags t ON t.id=dt.tagId WHERE dt.datasetId=?`,
+            db.knexInstance(), // TODO: pass down the knex instance instead of using the global one
```
Comment (sophiamersmann) [inline db/model/Dataset.ts:113]: Did you miss this or is this a todo for later?

File: adminSiteServer/apiRouter.ts [hunk 2]
```diff
@@ -1257,38 +1259,29 @@ apiRouter.post(
...
-            "updatedAt",
-            "lastLogin",
-            "lastSeen",
-        ],
-        order: { lastSeen: "DESC" },
-    }),
+    users: await db
+        .knexInstance()
+        .select(
+            "id" satisfies keyof DbPlainUser,
```
Comment (sophiamersmann) [inline adminSiteServer/apiRouter.ts:1265]: I didn't know about `satifies`. Neat!

**Review by sophiamersmann — APPROVED** [review]
Looks good to me!

Thanks again for finishing up the migrations :)

**Pull request merged**